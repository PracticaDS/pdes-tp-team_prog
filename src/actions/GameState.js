import {
  TICK,
  SELECT_MACHINE,
  SELECT_ACTION,
  SELECT_MOVE_BLOCK,
  DESELECT_MOVE_BLOCK,
  BUY_MACHINE,
  PLAY_GAME_BEGIN,
  PLAY_GAME_SUCCESS,
  PLAY_GAME_FAILURE,
  DELETE_GAME_BEGIN,
  DELETE_GAME_SUCCESS,
  DELETE_GAME_FAILURE,
  NEW_GAME_BEGIN,
  NEW_GAME_SUCCESS,
  NEW_GAME_FAILURE,
} from '../utils/actionTypes'
import {
  API_URL,
  createDeleteGameRequest,
  createNewGameRequest,
  createPlayGameRequest,
} from './utils'

export const createStackAction = ({ type, body }) => ({
  type,
  body: {
    ...body,
    stack: true,
  },
})

export const tick = () => ({
  type: TICK,
  body: {},
})

export const selectMachine = machine => ({
  type: SELECT_MACHINE,
  body: { machine },
})

export const selectAction = actionType => ({
  type: SELECT_ACTION,
  body: { actionType },
})

export const selectMoveBlock = node => ({
  type: SELECT_MOVE_BLOCK,
  body: { ...node },
})

export const deselectMoveBlock = () => ({
  type: DESELECT_MOVE_BLOCK,
  body: {},
})

export const buyMachine = price => ({
  type: BUY_MACHINE,
  body: { price },
})

const newGameBegin = () => ({
  type: NEW_GAME_BEGIN,
  body: {},
})
const newGameSuccess = ({ id }) => ({
  type: NEW_GAME_SUCCESS,
  body: { id },
})
const newGameFailure = ({ id }) => ({
  type: NEW_GAME_FAILURE,
  body: { id },
})

export const newGame = userId => dispatch => {
  dispatch(newGameBegin(userId))
  return fetch(`${API_URL}/users/${userId}/games/`, createNewGameRequest())
    .then(() => {
      // .then(res => res.json())
      // .then(({ result }) => {
      //   dispatch(newGameSuccess(result))
      dispatch(newGameSuccess({ id: userId }))
    })
    .catch(() => {
      // dispatch(newGameFailure(err))
      dispatch(newGameFailure({ id: userId }))
    })
}

export const playGameBegin = () => ({
  type: PLAY_GAME_BEGIN,
  body: {},
})
export const playGameSuccess = result => ({
  type: PLAY_GAME_SUCCESS,
  body: { ...result },
})
export const playGameFailure = result => ({
  type: PLAY_GAME_FAILURE,
  body: { ...result },
})

export const playGame = (userId, gameId) => dispatch => {
  dispatch(playGameBegin(gameId))
  return fetch(`${API_URL}/users/${userId}/games/${gameId}`, createPlayGameRequest(gameId))
    .then(res => res.json())
    .then(({ result }) => {
      dispatch(playGameSuccess(result))
    })
    .catch(err => {
      dispatch(playGameFailure(err))
    })
}

export const deleteGameBegin = () => ({
  type: DELETE_GAME_BEGIN,
  body: {},
})
export const deleteGameSuccess = result => ({
  type: DELETE_GAME_SUCCESS,
  body: { ...result },
})
export const deleteGameFailure = result => ({
  type: DELETE_GAME_FAILURE,
  body: { ...result },
})

export const deleteGame = (userId, gameId) => dispatch => {
  dispatch(deleteGameBegin(gameId))
  return fetch(`${API_URL}/users/${userId}/games/${gameId}`, createDeleteGameRequest(gameId))
    .then(res => res.json())
    .then(({ result }) => {
      dispatch(deleteGameSuccess(result))
    })
    .catch(err => {
      dispatch(deleteGameFailure(err))
    })
}
