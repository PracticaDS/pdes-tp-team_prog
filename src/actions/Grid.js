import {
  UPDATE_BLOCK,
  DELETE_BLOCK,
  CREATE_RAW_MATERIAL,
  CREATE_ITEMS,
  DELETE_ITEMS,
} from '../utils/actionTypes'

export const updateBlock = (position, machine) => ({
  type: UPDATE_BLOCK,
  body: { position, machine },
})

export const createRawMaterial = (position, material) => ({
  type: CREATE_RAW_MATERIAL,
  body: { position, material },
})

export const deleteBlock = position => ({
  type: DELETE_BLOCK,
  body: { position },
})

export const createItems = (position, items) => ({
  type: CREATE_ITEMS,
  body: { position, items },
})

export const deleteItems = (position, items) => ({
  type: DELETE_ITEMS,
  body: { position, items },
})
