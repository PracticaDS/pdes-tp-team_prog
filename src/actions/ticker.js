// @flow
/**
 * @module Actions/Ticker
 * @desc Tick Actions
 */
import { createActions } from 'redux-actions';

import { ActionTypes } from 'constants/index';

export const nextTick = createActions({
  [ActionTypes.NEXT_TICK]: () => ({}),
});