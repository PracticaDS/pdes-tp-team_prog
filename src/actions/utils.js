export const BEGIN = 'BEGIN'
export const SUCCESS = 'SUCCESS'
export const ERROR = 'ERROR'
export const GET = 'GET'
export const POST = 'POST'
export const PUT = 'PUT'
export const DELETE = 'DELETE'

export const customFetch = (api, req) =>
  new Promise((resolve, reject) => {
    try {
      setTimeout(() => {
        resolve(req)
      }, 1000)
    } catch (err) {
      reject(err)
    }
  })

export const createUpdateGameRequest = game => ({
  method: 'PUT',
  mode: 'cors',
  headers: {
    'Content-Type': 'application/json',
  },
  body: JSON.stringify(game),
})

export const API_URL = process.env.API_URL || 'http://localhost:8080/'

export const createAction = (type, body) => ({ type, body })

const createRequest = (payload, method) => ({
  method,
  mode: 'cors',
  headers: {
    'Content-Type': 'application/json',
  },
  ...(method !== 'GET' && payload && { body: JSON.stringify(payload) }),
})

export const createThunk = ({
  action,
  url,
  payload,
  parse = ({ result }) => result,
  method,
}) => dispatch => {
  dispatch(createAction(`${action}_${BEGIN}`))
  return fetch(`${API_URL}${url}`, createRequest(payload, method))
    .then(result => result.json())
    .then(result => {
      dispatch(createAction(`${action}_${SUCCESS}`, parse({ result, type: SUCCESS })))
    })
    .catch(result => {
      dispatch(createAction(`${action}_${SUCCESS}`, parse({ result, type: ERROR })))
    })
}
