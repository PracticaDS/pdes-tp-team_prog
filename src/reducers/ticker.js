import { handleActions } from 'redux-actions';
import immutable from 'immutability-helper';

import { STATUS, ActionTypes } from 'constants/index';

export const initialTicker = {
  tick: 0
};

export default {
  tick: handleActions(
    {
      [ActionTypes.NEXT_TICK]: state =>
        immutable(state, {
          tick: { $set: state.tick + 1 }
        }),
    },
    initialTicker,
  ),
};
