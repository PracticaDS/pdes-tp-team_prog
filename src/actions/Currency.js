import { RESTART_CURRENCY, INCREMENT_CURRENCY } from '../utils/actionTypes'

export const restartCurrency = () => ({
  type: RESTART_CURRENCY,
  body: {},
})

export const incrementCurrency = addedCurrency => ({
  type: INCREMENT_CURRENCY,
  body: { addedCurrency },
})
