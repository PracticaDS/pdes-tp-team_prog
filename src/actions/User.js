import { API_URL, createUserRequest } from './utils'
import { LOGIN_BEGIN, LOGIN_FAILURE, LOGIN_SUCCESS } from '../utils/actionTypes'

const loginBegin = user => ({
  type: LOGIN_BEGIN,
  body: { ...user },
})

const loginSuccess = user => ({
  type: LOGIN_SUCCESS,
  body: { ...user },
})

const loginFailure = user => ({
  type: LOGIN_FAILURE,
  body: { ...user },
})

export const login = user => dispatch => {
  dispatch(loginBegin(user))
  return fetch(`${API_URL}/user/login`, createUserRequest(user))
    .then(res => res.json())
    .then(({ result }) => {
      dispatch(loginSuccess(result))
    })
    .catch(err => {
      dispatch(loginFailure(err))
    })
}
