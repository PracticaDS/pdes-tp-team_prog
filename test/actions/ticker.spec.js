import { nextTick } from 'actions/ticker';

describe('Ticker Action', () => {
  it('nextTick should return an action', () => {
    expect(nextTick()).toMatchSnapshot();
  });
});
