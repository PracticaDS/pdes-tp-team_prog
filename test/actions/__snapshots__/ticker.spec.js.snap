// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker Action nextTick should return an action 1`] = `
Object {
  "payload": Object {},
  "type": "NEXT_TICK",
}
`;
